server/src/main/java/org/elasticsearch/index/engine/Engine.java

```diff
@@ -455,33 +455,64 @@ protected static final class IndexThrottle {
         private final CounterMetric throttleTimeMillisMetric = new CounterMetric();
         private volatile long startOfThrottleNS;
         private static final ReleasableLock NOOP_LOCK = new ReleasableLock(new NoOpLock());
-        private final PauseLock throttlingLock;
-        private final ReleasableLock lockReference;
+        private final ReleasableLock lockReference = new ReleasableLock(new ReentrantLock());
+        private final Lock pauseIndexingLock = new ReentrantLock();
+        private final Condition pauseCondition = pauseIndexingLock.newCondition();
+        private final ReleasableLock pauseLockReference = new ReleasableLock(pauseIndexingLock);
+        private volatile AtomicBoolean pauseThrottling = new AtomicBoolean();
```

Comment:
Can we name this and the method differently, like `suspendThrottling`? `pauseThrottling` sounds like a variable indicating that the "pause throttling" mechanism is enabled when instead it is that it is disabled ;-)